stores/chain_test.go

```diff
@@ -391,3 +393,74 @@ func TestProcessChainUpdate(t *testing.T) {
 		t.Fatal("expected ErrIndexMissmatch", err)
 	}
 }
+
+func TestContractElements(t *testing.T) {
+	ss := newTestSQLStore(t, defaultTestSQLStoreConfig)
+
+	// add test host and contract
+	hks, err := ss.addTestHosts(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fcids, _, err := ss.addTestContracts(hks)
+	if err != nil {
+		t.Fatal(err)
+	} else if len(fcids) != 1 {
+		t.Fatal("expected one contract", len(fcids))
+	}
+	fcid := fcids[0]
+
+	// declare some contract data - just enough to make sure it is persisted
+	contract := types.V2FileContract{
+		Capacity: 1,
+		Filesize: 2,
+	}
+
+	assertContractElement := func(tx sql.ChainUpdateTx, leafIndex uint64, proof []types.Hash256) {
+		t.Helper()
+		fce, err := tx.FileContractElement(fcid)
+		if err != nil {
+			t.Fatal(err)
+		} else if fce.ID != fcid {
+			t.Fatalf("unexpected contract id %v", fce.ID)
+		} else if fce.StateElement.LeafIndex != leafIndex {
+			t.Fatalf("unexpected leaf index %v", fce.StateElement.LeafIndex)
+		} else if !reflect.DeepEqual(fce.StateElement.MerkleProof, proof) {
+			t.Fatalf("unexpected merkle proof %v", fce.StateElement.MerkleProof)
+		} else if !reflect.DeepEqual(fce.V2FileContract, contract) {
+			t.Fatalf("unexpected contract %v", fce.V2FileContract)
+		}
+	}
+
+	// check current contract state
+	if err := ss.ProcessChainUpdate(context.Background(), func(tx sql.ChainUpdateTx) error {
+		// add a new contract element
+		err := tx.InsertFileContractElements([]types.V2FileContractElement{
+			{
+				ID: fcid,
+				StateElement: types.StateElement{
+					LeafIndex:   1,
+					MerkleProof: []types.Hash256{{1}},
+				},
+				V2FileContract: contract,
+			},
+		})
+		if err != nil {
+			return err
+		}
+		assertContractElement(tx, 1, []types.Hash256{{1}})
+
+		// TODO: update the element's proof
```

Comment:
leftover? we can already implement this test with what's currently available to us right?